### ISSUE TITLE
Dev branch tasks

### ISSUE THREAD

**Issue by nmoinvaz** [nmoinvaz/minizip#168]
Here are the tasks I have planned for the dev branch still. If anybody wants to pick up one of these tasks let me know.

1. ~~Need CMake projects for linux and for continuous integration. #163~~ 
2. ~~Add back compatibility layer.~~
3. Check into adding back HAVE_APPLE_COMPRESSION. Possibly adding a new "apple" stream.
4. ~~Add zipping of directories into minizip.c with the dir searching routines added into mz_strm_posix and mz_strm_win32.~~
5. ~~Check into version_madeby being defined per-platform in mz_strm.h.~~
6. ~~Add test python script that tests minizip and miniunzip against various compress methods and levels and encryption methods.~~
7. ~~Check into adding a disk spanning stream.~~
8. ~~Update the test python script to download the git repository and zip/unzip it for its tests, now that 4 has been completed.~~
9. ~~Update Linux Qbs project files.~~
10. ~~Investigate arc4random. #175~~

**Comment by nmoinvaz** [comment]
1, 4, and 6 are complete.

**Comment by nmoinvaz** [comment]
For 2, I looked into adding a compatibility layer and it is too much work because of the old IO API layer that 1.2 had.

**Comment by Coeur** [comment]
Bah, if there is no compatibility layer, we better be ready to maintain two branches in parallel for some time (maybe labelling the current master as "1.2" or "legacy" branch).

I do plan to make a compatibility layer, just I'm a bit slow to code compare to you.

**Comment by nmoinvaz** [comment]
That would be great. I recommend starting from the point where I left off..

https://github.com/nmoinvaz/minizip/blob/41cfdcb45fa0277a326b592853cae0383acf47ba/mz_compat.c
https://github.com/nmoinvaz/minizip/blob/41cfdcb45fa0277a326b592853cae0383acf47ba/mz_compat.h

**Comment by nmoinvaz** [comment]
When I make the switch, I do plan on tagging the old branch.

**Comment by Coeur** [comment]
I don't mean a tag, I mean a branch, where commits can be added.

Note that I failed at making a satisfying compatibility layer, so I will postpone my plans to integrate MiniZip2.

**Comment by nmoinvaz** [comment]
Ok. I see now that not having the compatibility layer is blocking adoption, so I took some time and figured out how to do it. Let me know if you have any issues with it.

**Comment by Coeur** [comment]
What about #180 and #181?

**Comment by nmoinvaz** [comment]
I have completed 8.

**Comment by Coeur** [comment]
point 10 (arc4random) is not needed to be done _before_ merging to master: it can be done after, as it's not a regression (there is the same consideration on current master branch).

**Comment by Coeur** [comment]
I could be wrong, there might have been a bug, but I remember that Apple compression was awful... compressed filesize being bigger than the original.

We should test it a bit (with the master code) to see if it's really worth it.

**Comment by Coeur** [comment]
I'm starting to wonder if you're going to do a `libgladmanaes` as well, ah ah.

**Comment by nmoinvaz** [comment]
I'm not sure.. all these projects are getting pretty hairy to manage. I am thinking of going CMakeLists.txt only - and using cmake will generate the project files for win, macOS, and linux automatically. Thoughts?

**Comment by nmoinvaz** [comment]
As far as apple compression, I'm not really sure what it buys anybody except a smaller executable (at the cost of maintaining more code associated with it). I wouldn't be surprised if it was just using stock zlib & liblzma underneath.

**Comment by nmoinvaz** [comment]
I've decided to only support cmake going forward. It can generate the projects on the various platforms. And you can tell it what features to use. Number 9 is now obsolete.

**Comment by nmoinvaz** [comment]
Completed 10. Added writing/zipping for disk splitting. Still have to do reading/unzipping.

**Comment by nmoinvaz** [comment]
Finished the disk splitting stream.

**Comment by nmoinvaz** [comment]
Don't really care about the apple compression for now.

**Comment by nmoinvaz** [comment]
I have created the 1.2 branch and also pushed dev to master.
